code-actions-sample/src/diagnostics.ts

```diff
@@ -23,21 +23,27 @@ export function refreshDiagnostics(doc: vscode.TextDocument, emojiDiagnostics: v
 
 	for (let lineIndex = 0; lineIndex < doc.lineCount; lineIndex++) {
 		const lineOfText = doc.lineAt(lineIndex);
-		if (lineOfText.text.includes(EMOJI)) {
-			diagnostics.push(createDiagnostic(doc, lineOfText, lineIndex));
+		const regex = RegExp(EMOJI, 'g');
```

Comment:
Use `new`

Also this should escape special regex character in `EMOJI` in case someone modifies the code: https://stackoverflow.com/questions/3446170/escape-string-for-use-in-javascript-regex